app/lib/steps/applicant_details_section.rb

```diff
@@ -18,12 +18,16 @@ def grouped_steps_for(session_data)
       end
 
       def groups(session_data)
-        [:applicant, dependant_details(session_data)].compact
+        [:applicant, dependant_details(session_data), partner_details(session_data)].compact
       end
 
       def dependant_details(session_data)
         :dependant_details unless Steps::Logic.passported?(session_data)
       end
+
+      def partner_details(session_data)
+        :partner_details if Steps::Logic.passported?(session_data) && FeatureFlags.enabled?(:household_section)
```

Comment:
If the client doesn't have a partner this method should return nil - otherwise we see the partner details screen even when there is no partner.